### PR TITLE
[Inductor] Pass device interface to the worker compile

### PR DESCRIPTION
**Pull request by matthewhagraphcore**
Summary: In `codecache.py` pass the device_interface directly to `_worker_compile()` instead of calling `get_device_interface()` from inside the function.

If the device_interface is registered by an out-of-tree module then it will only be registered inside the main process and not inside the worker process. This fixes this issue. Happy to add a test if required. 

Test plan:
No tests added

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

**Comment by ezyang** [comment]
lint

**Comment by matthewhagraphcore** [comment]
`warning: "VLOG_IS_ON" redefined` in [linux-focal-py3_8-clang9-xla](https://github.com/pytorch/pytorch/actions/runs/8389990657/job/22993709883?pr=122492#logs) . Unrelated failure?

File: torch/_inductor/codecache.py [hunk 1]
```diff
@@ -2709,7 +2713,7 @@ def triton(
             device = torch.device(device_str, device_interface.current_device())
             cc = device_interface.get_compute_capability(device)
             future = self.process_pool().submit(
-                _worker_compile, kernel_name, source_code, cc, device
+                _worker_compile, kernel_name, source_code, cc, device, device_interface
```
[the remark this answers is not in the document]
Comment (matthewhagraphcore) [reply torch/_inductor/codecache.py:2762]: OK, I've updated. It was expecting a type, not an instance.

**Pull request closed without merging**